Chemistry mucking

comma opened the pull request:
Decoupling consumed dose from reagent into mob  …
It will no longer immediately reset once chemical in bloodstream is done processing.
It will tickle down at double metabolism rate.

In effect currently this buffs alcohol, since wait times between sips are now longer for non-drinking drinking.
It also buffs SUGAR and HONEY againt unathi, as it goes off the dose too.
Technically consequitive doses of soporific will KO you too faster, but what's stopping people from just dosing you high in the first place

System consumed dosages can be detected by handheld scanners for scannable chems, adv masspectromers for others.

Slight handle_chemicals_in_body refactor  …
Moved everything not related to actual chemicals into vaarious placees, species procs, organs processing, regular status updates. Merges trace chem handling into it.

Axed thing for pre-brainmed defibs, no need to proces chems in dead now.

comma committed:
Decoupling consumed dose from reagent into mob

It will no longer immediately reset once chemical in bloodstream is done processing.
It will tickle down at double metabolism rate.

In effect currently this buffs alcohol, since wait times between sips are now longer for non-drinking drinking.
It also buffs SUGAR and HONEY againt unathi, as it goes off the dose too.
Technically consequitive doses of soporific will KO you too faster, but what's stopping people from just dosing you high in the first place

System consumed dosages can be detected by handheld scanners for scannable chems, adv masspectromers for others.

comma commented:
Well it's not reall WIP since i don't relaly ahve anything to add, just need to test it now tha I'm not on a craptop. Feel free to poke at stuff

comma commented:
Okay as ready as it gets

PsiOmegaDelta commented on code/modules/mob/living/carbon/human/life.dm:
```diff
@@ -732,6 +646,17 @@
 		if(gloves && germ_level > gloves.germ_level && prob(10))
 			gloves.germ_level += 1
 
+		if(vsc.plc.CONTAMINATION_LOSS && !(status_flags & GODMODE)) 
```
We'll never have to check for the GODMODE flag again, will we? 
Seems rather final at https://github.com/Baystation12/Baystation12/pull/18759/files#diff-2de2df982739a4fbdb54a4b2715d6f92R516.

comma replied:
well gods are omnipotent so who knows

PurpleMartinJCK commented on code/modules/mob/living/carbon/human/species/station/station.dm:
```diff
@@ -374,3 +374,59 @@
+		H.take_overall_damage(2,0)
+		//traumatic_shock is updated every tick, incrementing that is pointless - shock_stage is the counter.
+		//Not that it matters much for diona, who have NO_PAIN.
+		H.shock_stage++
```
Since this is species specific now, shock is redundant.

PurpleMartinJCK commented on code/modules/organs/external/diona.dm:
```diff
@@ -244,5 +244,22 @@
+			light_amount =  5
+		light_amount = T.get_lumcount() * 10
+	owner.nutrition   += light_amount
+	owner.shock_stage -= light_amount
```
Shock stage here should be removed too.

comma replied:
What if you implant this organ into a mob that does feel pain :3 
It's harmless to diona since for NO PAIN species shock stage is reset to 0 every tick, but would be nice for that hypothetical case.

PurpleMartinJCK commented on code/modules/organs/external/diona.dm:
```diff
@@ -244,5 +244,22 @@
+			light_amount = max(0, min(14,(L.lum_r + L.lum_g + L.lum_b)) * 1.5) //hardcapped so it's not abused by having a ton of flashlights
+		else
+			light_amount =  5
+		light_amount = T.get_lumcount() * 10
```
You go through the effort of finding light_amount above, and now just overwrite it here.

comma committed:
Slight handle_chemicals_in_body refactor

Moved everything not related to actual chemicals into vaarious placees, species procs, organs processing, regular status updates. Merges trace chem handling into it.

Axed thing for pre-brainmed defibs, no need to proces chems in dead now.

PurpleMartinJCK commented on code/modules/mob/living/carbon/human/species/station/station.dm:
```diff
@@ -374,3 +374,56 @@
+		if(H.getToxLoss())
+			H.adjustToxLoss(-8)
+			H.nutrition -= 2
+		if(H.getOxyLoss())
```
They don't have oxyloss anymore as it's hard coded out.

PurpleMartinJCK commented on code/modules/mob/living/carbon/human/species/station/station.dm:
```diff
@@ -374,3 +374,56 @@
+		if(H.getFireLoss())
+			H.adjustFireLoss(-4)
+			H.nutrition -= 2
+		if(H.getToxLoss())
```
Now that toxloss is in organs. it's healed below on line 406.

PurpleMartinJCK commented on code/modules/mob/living/carbon/human/life.dm:
```diff
@@ -732,6 +646,17 @@
 		if(gloves && germ_level > gloves.germ_level && prob(10))
 			gloves.germ_level += 1
 
+		if(vsc.plc.CONTAMINATION_LOSS) 
```
This needs to be made species specific or use a species tag so that species that aren't damaged by phoron aren't affected.

comma replied:
Way beyond what I am doing here

PurpleMartinJCK replied:
We talked 👌

PsiOmegaDelta replied:
And this is why discussions should be done on Github and not in any other random place that isn't remotely connected to comments made here.

PurpleMartinJCK replied:
What do you mean?

comma replied:
True.
Anway gist of it is that I just moved stuff around, I didn't sign up for adding species features for it, cue not doing this.

PurpleMartinJCK commented on code/modules/organs/external/diona.dm:
```diff
@@ -244,5 +244,17 @@
+	var/light_amount = 0 //how much light there is in the place, affects receiving nutrition and healing
+	if(isturf(owner.loc)) //else, there's considered to be no light
+		var/turf/T = owner.loc
+		light_amount = T.get_lumcount() * 10
```
How does this function on unsimulated turf like in the holodeck?

comma replied:
it's base turf level, using llighting overlays

PurpleMartinJCK replied:
We also talked 👌

comma replied:
My comment p much was that, it uses pretty much same calculations that I deleted.
